public/app/features/alerting/unified/components/rule-editor/query-and-alert-condition/QueryAndExpressionsStep.tsx

```diff
@@ -167,14 +167,20 @@ export const QueryAndExpressionsStep = ({ editingExistingRule, onDataChange }: P
   const isCloudAlertRuleType = isCloudAlertingRuleByType(type);
   const [showResetModeModal, setShowResetModal] = useState(false);
 
+  const simplifiedQueryInForm = editorSettings?.simplifiedQueryEditor;
+
   const { isAdvancedMode, simpleCondition, setSimpleCondition } = useAdvancedMode(
-    editorSettings,
+    simplifiedQueryInForm,
     isGrafanaAlertingType,
     isNewFromQueryParams,
     dataQueries,
     expressionQueries
   );
 
+  useEffect(() => {
+    setValue('editorSettings.simplifiedQueryEditor', !isAdvancedMode);
```

Comment:
I think we should populate correct value in the `formValuesFromQueryParams` function instead

Reply:
with the last commit, we don't need this useEffect anymore